xbmc/video/VideoDatabase.cpp

```diff
@@ -7890,10 +7892,14 @@ void CVideoDatabase::CleanDatabase(CGUIDialogProgressBarHandle* handle, const st
       std::map<int, bool>::const_iterator pathsDeleteDecision = pathsDeleteDecisions.find(m_pDS->fv(0).get_asInt());
       // Check if we have a decision for the parent path
       std::map<int, bool>::const_iterator pathsDeleteDecisionByParent = pathsDeleteDecisions.find(m_pDS->fv(2).get_asInt());
-      if (((pathsDeleteDecision != pathsDeleteDecisions.end() && pathsDeleteDecision->second) ||
-           (pathsDeleteDecision == pathsDeleteDecisions.end() && !CDirectory::Exists(m_pDS->fv(1).get_asString(), false))) &&
-          ((pathsDeleteDecisionByParent != pathsDeleteDecisions.end() && pathsDeleteDecisionByParent->second) ||
-           (pathsDeleteDecisionByParent == pathsDeleteDecisions.end())))
+      bool hasDecision = pathsDeleteDecision != pathsDeleteDecisions.end();
+      bool hasDecisionForParent = pathsDeleteDecisionByParent != pathsDeleteDecisions.end();
+      std::string parentPath = m_pDS->fv(3).get_asString();
+
+      if (((hasDecision && pathsDeleteDecision->second) ||
```

Comment:
I think it's time to add some comments what this is supposed to do:
```
// add the path to the list of paths to be deleted if
//   it has been handled before and should be deleted
//   OR
//   it hasn't been handled before and doesn't exist
// AND
//   its parent path has been handled before
//   AND
//     should be deleted
//     OR
//     doesn't exists
//   its parent path hasn't been handled before
```
Reading it like that I'm actually a bit confused about the AND combination of the path and the parentpath parts.